### ISSUE TITLE
keep programs list open after move/delete operations

### ISSUE THREAD

**Issue by gitterman** [Open-Shell/Open-Shell-Menu#238]
After migrating from windows7 with classic shell to openshell my programs menu always closes when I rearrange something on the menu.
I could not find a setting to keep it open after a move or delete operation.
Reason>
I want to keep my programs menu cleanly structured, so after every program installation I move the main program exe to a suitable subfolder and eventually delete the flat program folder which often just contain superfluous shortcuts.
Thus it is very annoying to re-open START - Programs every time again after a move or delete operation on the programs menu.

**Comment by Ibuprophen** [comment]
This is a very unique/interesting feature request that I do experience, though I never really thought about before. 

It looks like it's tied to a type of "focus" for Windows. In another words, after performing something like moving, adding, etc..., the Windows OS looses it's focus/attention to the start menu (that would keep the start menu open). 

It looks, to me, like Windows and the Open Shell Software doesn't accommodate this. 

I feel that this feature would be something that should be added to the Open Shell customization settings for individuals to be able to decide to Enable/Disable this feature. 

I hope that I had explained the above okay via text... 


~Ibuprophen

**Comment by ge0rdi** [comment]
@gitterman 
Can you possibly create some gif animation of what are you doing exactly?
Or at least describe it step-by-step.
Also what OS version do you have?

When I'm rearranging items in `Programs`, menu stays opened and I don't need to reopen it again.

**Comment by Ibuprophen** [comment]
@ge0rdi, It does close for me after performing a move as well as creating a new folder. I just never really gave it much thought to it remaining open. 

This was regarding my PC with Windows 8.1 Pro and one of my Laptops with Windows 10 (just barely got going). 


~Ibuprophen

**Comment by gitterman** [comment]
Sorry for not being precise. 
I did re-install my w7 desktop computer with w10 this week and I am in the process of getting all my programs installed and configured again.
When I move or delete links from the start menu and windows intercepts with a dialog box then the start menu gets closed and I need to open it again.
It would be nice if the menu could stay open in these cases.
On my legacy w7 installation I had switched off the delete confirmation of the recycle bin and at least in this case I could delete items from the start menu without getting it closed.
On my w10 with openshell though the start menu always gets closed.

**Comment by ge0rdi** [comment]
> When I move or delete links from the start menu and windows intercepts with a dialog box then the start menu gets closed and I need to open it again.

Ok, now I'm able to reproduce it.

I was able to modify `OpenShell` to not close menu on these actions. But now I'm trying to evaluate whether it will have some negative consequences or not.
Will create PR eventually, so you will be able to test it too.

**Comment by gitterman** [comment]
@ge0rdi
Thanks a lot, that sounds very promising. 
I am looking forward to test it too.

**Comment by ge0rdi** [comment]
@gitterman 
Please, try with this build:
https://ci.appveyor.com/project/passionate-coder/open-shell-menu/builds/28128984/artifacts

**Comment by gitterman** [comment]
@ge0rdi
Great work,
I have tested the new build for 2 days now and it works like a charm.
So far I did not experience any problems.

**Comment by ge0rdi** [comment]
@gitterman 
Glad to hear that :)

**Comment by gitterman** [comment]
@ge0rdi
yesterday I updated windows to 1903 and now I have a very strange behavior with open shell.
My system is US english but I am located in Germany and I do have set time and date to German format.
After the windows update the windows start menu is still English but the open shell menu shows German entries. I checked the menu language and it still is English US. I even tried to set it to German and then revert to English but still it shows a weird mixture of German and English text (see attachment)
Is this a known issue or may it be due to the special build I am running?
What is the best way to get back a clean English menu without loosing all my settings?
I
![2019-10-25 09_36_21-keep programs list open after move_delete operations · Issue #238 · Open-Shell_O](https://user-images.githubusercontent.com/10762929/67552374-4375b800-f70b-11e9-87ef-ef03fd8757a4.png)

**Comment by ge0rdi** [comment]
> Is this a known issue or may it be due to the special build I am running?

Didn't see such issue yet.
And I don't think it is related to the latest build.

> What is the best way to get back a clean English menu without loosing all my settings?

You can just install older version and check whether it will be the same or not.
You may need to uninstall current version though, but don't worry settings are not removed on uninstall.

**Comment by gitterman** [comment]
Thanks for the quick response.
I uninstalled open-shell and the problem is still there.
I opened the settings applet and although initially English some of the sub-applets appeared in German. 
So I think the crappy M$ update has done something totally weird with my language settings.
I am going to uninstall German as an additional language and rebuild from that.

**Comment by ge0rdi** [comment]
Fixed in latest nightly build.

**Comment by coddec** [comment]
https://github.com/Open-Shell/Open-Shell-Menu/pull/240#issuecomment-542323474

**Comment by Gittyperson** [comment]
I tried deleting a few shortcuts today directly from the Start Menu. It was impossible, because the delete confirmation was inaccessible, hidden behind the menu which was on top. No way to access the prompt. Using "Classic with two columns" which can take a lot of screen space.

What's more, the open menu **will not go away** -locked by the confirmation dialog?- and a **sign-out** operation is required. This is quite a significant issue. Not even killing StartMenu.exe worked.

I know that displaying the delete confirmation prompt is not the default any more In recent Windows versions (awful decision imo) but the does not mean Open-Shell should interfere with basic Windows operations. If the OP frequently moves/deletes/organizes shortcuts, he should do it directly in the Start Menu folders (ProgramData, Roaming). The Start Menu isn't really meant for that.

So, please reconsider this or add the option to make the Start Menu disappear as it always has since... ever.

OS 4.4.142, W10 1909 18363.476

**Comment by Gittyperson** [comment]
The only workaround I found, was to use Alt-Tab to close the confirmation prompt, without of course deleting the shortcut (which has now become seemingly impossible when using confirmation prompts). Only after closing the confirmation prompt, will the Start Menu close.

**Comment by ge0rdi** [comment]
@Gittyperson 
Please, create new issue for this.
Eventually add screenshot that demonstrates the issue (I assume your start menu is quite big).

There is no point to discuss it in already closed issue.

**Comment by gitterman** [comment]
I can confirm the issue. If I move the "delete confirmation" window behind the open start menu it becomes inaccessible but it can be closed with ALT-TAB. 
The question to me is: why is the start menu on top of the dialog box!

@Gittyperson: You could also delete an entry if you right click the parent folder of the item which you want to delete and choose "open" to open an explorer. This closes the start menu and you can delete your desired item from the explorer window without any problems.

**Comment by Gittyperson** [comment]
Thanks gitterman but that's a workaround, not really a solution. The [issue](https://github.com/Open-Shell/Open-Shell-Menu/issues/257) remains as Windows dialogs should not be blocked. As mentioned earlier, I prefer to edit shortcuts directly inside their Windows folders anyway, not using the Start Menu.
